pkg/source/reconciler/mtsource/kafkasource.go

```diff
@@ -108,7 +110,11 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1beta1.KafkaSource
 
 func (r *Reconciler) reconcileMTReceiveAdapter(src *v1beta1.KafkaSource) error {
 	placements, err := r.scheduler.Schedule(src)
-
+	sort.SliceStable(placements, func(i int, j int) bool {
+		podIOrdinal, _ := strconv.Atoi(placements[i].PodName[len(placements[i].PodName)-1:])
+		podJOrdinal, _ := strconv.Atoi(placements[j].PodName[len(placements[j].PodName)-1:])
+		return podIOrdinal < podJOrdinal
+	})
```

Comment:
`Schedule` should return a sorted list on placements as it relies on it.

Reply:
The current logic of addVReplica is adding `vreplicas` to existing pod, then allocate additional pods. In case existing pod is not a sequence, it's possible scheduler adds `vreplicas` to a pod that's in between existing pods. In this scenario, Schedule won't return a sorted list.